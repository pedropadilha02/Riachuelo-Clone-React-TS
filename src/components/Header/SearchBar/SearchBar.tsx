import styled from 'styled-components';
import { useProduct } from '../../../context/ProductContext';

const SearchStyled = styled.div`
  position: relative;

  @media (max-width: 1200px) {
    box-sizing: border-box;
  }
`;

const SearchIcon = styled.img`
  width: 32px;
  opacity: 26%;
  top: 8px;
  position: absolute;
  padding-left: 10px;
  color: rgba(0, 0, 0, 0.26);
  line-height: 40px;
  margin: auto 0;
`;

const SearchInput = styled.input`
  height: 40px;
  width: 100%;
  border-radius: 4px;
  border: 1px solid rgba(0, 0, 0, 0.23);
  padding-left: 40px;

  &:hover {
    border-color: #000;
  }

  &::placeholder {
    font-size: 0.9375rem;
    font-weight: 600;
    font-family: 'Open Sans';
    color: #9b9b9b;
  }
`;

export function SearchBar() {
  const { text, setText } = useProduct();

  function searchProducts(event: React.ChangeEvent<HTMLInputElement>) {
    setText(event.target.value);
  }

  return (
    <SearchStyled>
      <SearchIcon src="http://localhost:3000/assets/search.svg" />
      <SearchInput
        type="search"
        placeholder="O que você está procurando?"
        value={text}
        onChange={searchProducts}
      />
    </SearchStyled>
  );
}
