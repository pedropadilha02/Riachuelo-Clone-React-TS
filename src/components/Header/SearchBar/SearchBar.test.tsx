import { render } from '@testing-library/react';
import { SearchBar } from './SearchBar';

describe('Teste de renderização do SearchBar', () => {
  it('O snapshot do component deve permanecer semre o mesmo', () => {
    const { container } = render(<SearchBar />);

    expect(container).toMatchSnapshot();
  });
});
