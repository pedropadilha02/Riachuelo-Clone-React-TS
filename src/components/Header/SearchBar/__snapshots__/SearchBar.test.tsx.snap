// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teste de renderização do SearchBar O snapshot do component deve permanecer semre o mesmo 1`] = `
<div>
  <div
    class="sc-bdnxRM tTQqD"
  >
    <img
      class="sc-gtsrHT TSJlF"
      src="http://localhost:3000/assets/search.svg"
    />
    <input
      class="sc-dlnjwi kCGmry"
      placeholder="O que você está procurando?"
      type="search"
      value=""
    />
  </div>
</div>
`;
